code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -5,49 +5,44 @@
 	anchored = TRUE
 	w_class = WEIGHT_CLASS_BULKY
 	canhear_range = 2
-	var/number = 0
-	var/anyai = 1
-	var/mob/living/silicon/ai/ai = list()
-	var/last_tick //used to delay the powercheck
 	dog_fashion = null
-	var/unfastened = FALSE
+	unscrewed = FALSE
 
 /obj/item/radio/intercom/unscrewed
-	unfastened = TRUE
+	unscrewed = TRUE
 
 /obj/item/radio/intercom/Initialize(mapload, ndir, building)
 	. = ..()
 	if(building)
 		setDir(ndir)
-	START_PROCESSING(SSobj, src)
-
-/obj/item/radio/intercom/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	return ..()
+	var/area/current_area = get_area(src)
+	if(!current_area)
+		return
+	RegisterSignal(current_area, COMSIG_AREA_POWER_CHANGE, .proc/AreaPowerCheck)

```

Comment:
Correct me if I'm wrong, but won't this break if you undo an intercom and move it to an entirely new area?

It'll still update...just not when the area its in updates---but the area it initialized in.

Reply:
It actually won't because the frame items you can pickup and move are `/obj/item/wallframe/intercom` types. When you place them on walls, it initializes the `/obj/item/radio/intercom`, which is the usable version that you can talk into/hear from. And when you take an intercom off the wall it actually gets destroyed.